### ISSUE TITLE
[Feature Request] Add ability to auto-maximize video to viewport dimensions on load

### ISSUE THREAD

**Issue by sjclayton** [YouTube-Enhancer/extension#163]
Title pretty much sums it up, would like a way to have the maximize behavior that is triggered by the maximize button to occur when a video is initially loaded. Having a toggle for this in settings page would be awesome.

Thanks.

**Comment by VampireChicken12** [comment]
@sjclayton Thanks for making an issue. I'll add that in the next update.

**Comment by VampireChicken12** [comment]
Hi @sjclayton,

While working on your requested feature, I noticed that the code for maximizing the player needs some cleaning up to smoothly integrate with the automatic maximize player feature. This cleanup might extend the time needed to finish your feature. Thanks for your understanding! 🛠️

**Comment by happygolucky919** [comment]
If I could make an addendum to this feature. Could you also make it so you can still scroll the page with Maximize enabled? It was the one feature I liked most about another youtube addon before it broke.

**Comment by VampireChicken12** [comment]
@happygolucky919 I should be able to do that I think.

**Comment by VampireChicken12** [comment]
Feature menu is offset above the player controls way more than it should be when maximize player is enabled

![Image](https://github.com/VampireChicken12/youtube-enhancer/assets/45531575/351cc12a-ccd0-4941-bd95-5476f1c22425)

**Comment by VampireChicken12** [comment]
![maximize-info](https://github.com/VampireChicken12/youtube-enhancer/assets/45531575/6bd2329a-1209-4ccc-96bb-6c32adec87ff)

Just a bit of info I put into a file so I know what I have to do when re-writing the maximize player feature.

I'm working on making the feature actually proper right now it works but it relies on the theater mode being enabled which is janky. The list of things to be updated  is what I have discovered that need adjusting ones with check marks I already know how to implement.

@sjclayton mentioning you to let you know I'm working on this the auto maximize feature shouldn't be delayed much longer after I finish determine what needs to be adjusted then implement that.

**Comment by VampireChicken12** [comment]
There is hope @sjclayton https://github.com/VampireChicken12/youtube-enhancer/commit/f97aa3bdbc9b4755861dbdfcb0e96dba4a473d39 sorry for taking so long. I just want to make the feature as good as possible.

**Comment by sjclayton** [comment]
@VampireChicken12 

Any hope for this? It's one of the main things that is still missing for me in using this extension full time.

**Comment by VampireChicken12** [comment]
@sjclayton I'm going to continue working on this. Are you okay with the seek preview being messed up until I figured out how to get it fixed. I was re-writing the maximize feature then I was going to implement this feature.

**Comment by sjclayton** [comment]
@VampireChicken12 

The seek preview is working fine for me currently as far as I know... I'm using the last actual release as of now (1.26.0)

**Comment by VampireChicken12** [comment]
@sjclayton 
yes in the current version it is but in the re-worked feature it doesn't work properly it. I can't show how it looks right now.

**Comment by sjclayton** [comment]
@VampireChicken12 

Well keep working on it and release it when it's ready... I can't not have the seek preview working.

**Comment by VampireChicken12** [comment]
@sjclayton okay, it works but it doesn't fit the the full size.

**Comment by VampireChicken12** [comment]
@sjclayton Sorry this took forever, recently with the new layout changes I had to work on the maximize feature anyways so I dialed in and got it done.
